core/scheduler/scheduler_test.go

```diff
@@ -312,6 +346,34 @@ func TestScheduler_GetDuty(t *testing.T) {
 	require.NoError(t, sched.Run())
 }
 
+// delayer implements scheduler.delayFunc and records the deadline and returns it immediately.
+type delayer struct {
+	mu        sync.Mutex
+	deadlines map[core.Duty]time.Time
+}
+
+func (d *delayer) Get() map[core.Duty]time.Time {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	return d.deadlines
+}
+
+// Delay implements scheduler.delayFunc and records the deadline and returns it immediately.
+func (d *delayer) Delay(duty core.Duty, deadline time.Time) <-chan time.Time {
+	d.mu.Lock()
```

Comment:
Is this method used or aimed to be used outside the package?

Reply:
see comment above

Reply:
ok made it unexported.